### PR TITLE
Add Grafana HTTP API client

### PR DESCRIPTION
**Pull request by lukaszcl**

### CODE REVIEW

**Review by AnieeG — COMMENTED** [review]
how about adding some tests?

File: grafana/client.go [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+	Annotations  map[string]string `json:"annotations"`
+}
+
+func (p ProvisionedAlertRule) String() string {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("ID: %d\n", p.ID))
+	sb.WriteString(fmt.Sprintf("UID: %s\n", p.UID))
+	sb.WriteString(fmt.Sprintf("FolderUID: %s\n", p.FolderUID))
+	sb.WriteString(fmt.Sprintf("Title: %s\n", p.Title))
+	// sb.WriteString("Data: [\n")
```
Comment (skudasov) [inline grafana/client.go:232]: Should we remove this?

File: grafana/client.go [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+}
+
+// ruler API is deprecated https://github.com/grafana/grafana/issues/74434
+func (g *Client) GetAlertsForDashboard(dashboardUID string) (map[string][]interface{}, *resty.Response, error) {
+	var result map[string][]interface{}
+	r, err := g.resty.R().SetResult(&result).Get("/api/ruler/grafana/api/v1/rules?dashboard_uid=" + url.QueryEscape(dashboardUID))
+	return result, r, err
+}
+
+type AlertQuery interface{}
```
Comment (skudasov) [inline grafana/client.go:211]: Do we plan to add more on top of it? Empty type.

**Review by skudasov — APPROVED** [review]

File: grafana/client.go (hunk 2 above)
Comment (lukaszcl) [reply grafana/client.go]: @skudasov I removed this type for now

**Review by skudasov — APPROVED** [review]

**Comment by lukaszcl** [comment]
> how about adding some tests?

@AnieeG this PR contains API client for external server (Grafana). We discussed it with Sergey and we do not see any tests that would be useful for it at this point.

**Pull request merged**